phoenix-pherf/src/main/java/org/apache/phoenix/pherf/workload/MultiThreadedRunner.java

```diff
@@ -88,13 +90,21 @@
      */
     @Override
     public Void call() throws Exception {
-        LOGGER.info("\n\nThread Starting " + threadName + " ; " + query.getStatement() + " for "
-                + numberOfExecutions + "times\n\n");
+        LOGGER.info("\n\nThread Starting " + threadName + " ; '" + query.getStatement() + "' for "
+                + numberOfExecutions + " times\n\n");
         Long start = EnvironmentEdgeManager.currentTimeMillis();
-        for (long i = numberOfExecutions; (i > 0 && ((EnvironmentEdgeManager.currentTimeMillis() - start)
-                < executionDurationInMs)); i--) {
+        for (long i = 0; i < numberOfExecutions; i++) {
+
+            if ((EnvironmentEdgeManager.currentTimeMillis() - start) >= executionDurationInMs) {
```

Comment:
Again here we should be calling `EnvironmentEdgeManager.currentTimeMillis() - start` once only, otherwise you're checking one thing and logging something slightly different. Maybe just do:

```java
long timeElapsed = EnvironmentEdgeManager.currentTimeMillis() - start
```
and use `timeElapsed` for both the if condition check and the log line